user/tests/wiring/ble/heartrate/application.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ ******************************************************************************
+  Copyright (c) 2015 Particle Industries, Inc.  All rights reserved.
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this program; if not, see <http://www.gnu.org/licenses/>.
+  ******************************************************************************
+ */
+
+/* Includes ------------------------------------------------------------------*/
+#include "application.h"
+
+BLEAttributeClass heartrate("heartrate", READ | NOTIFY);

```

Comment:
Please remove "Class" from the class name — it should just be `BLEAttribute`  like you had it before. I'm curious why you added "Class".

Reply:
I defined a new type `typedef BLEAttribute* BLEAttribute;` to maintain an attribute in peer device using a pointer. An peer attribute is associated with connection status, when disconnected the attribute should not be valid any more. If let user have a copy of peer attribute, then the copied attribute cannot be updated internally.